src/Test.UnitTests.Sarif/Core/AddressTests.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Numerics;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif;
+
+using Newtonsoft.Json;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Core
+{
+    public class AddressTests
+    {
+        private const string AbsoluteAddress = "\"AbsoluteAddress\"";
+        private static readonly TestAssetResourceExtractor s_extractor = new TestAssetResourceExtractor(typeof(AddressTests));
+
+        [Fact]
+        public void Address_VerifyRoundTripFromObject()
+        {
+            // Arrange
+            var address = new Address();
+
+            Address_VerifyRoundTripFromObjectHelper(address,
+                expectedShouldSerializeAbsoluteAddress: false, expectedAbsoluteAddress: -1);
+
+            // Arrange
+            address.AbsoluteAddress = long.MinValue;
+
+            Address_VerifyRoundTripFromObjectHelper(address,
+                expectedShouldSerializeAbsoluteAddress: false, expectedAbsoluteAddress: -1);
+
+            // Arrange
+            address.AbsoluteAddress--;
+
+            Address_VerifyRoundTripFromObjectHelper(address,
+                expectedShouldSerializeAbsoluteAddress: false, expectedAbsoluteAddress: -1);
```

Comment:
we have a lot of duplication here.

can't you create a struct/object/class that contains the initial values and do a foreach just like all the other places?

Reply:
@eddynaka and @shaopeng-gh: Something along these lines?
```// Arrange
var testCases = new[]
{
    new {
        absoluteAddress = <>,
        expectedShouldSerializeAbsoluteAddress = <>
    },
    etc.
}

var address = new Address();

foreach(var testCase in testCases)
{
    address.AbsoluteAddress = testCase.absoluteAddress;
    int expectedAbsoluteAddress = testCase.absoluteAddress < 0 ? -1 : absoluteAddress;
    // Act..Assert (combined in a single helper method)
    Address_VerifyRoundTripFromObjectHelper(
        address,
        testCase.expectedShouldSerializeAbsoluteAddress, 
        expectedAbsoluteAddress 
        
}```

Reply:
updated to use loops.